fuse/nodefs/fsconnector.go

```diff
@@ -126,9 +135,23 @@ func (c *FileSystemConnector) forgetUpdate(nodeID uint64, forgetCount int) {
 
 	if forgotten, handled := c.inodeMap.Forget(nodeID, forgetCount); forgotten {
 		node := (*Inode)(unsafe.Pointer(handled))
+		if node.IsDir() && len(node.Children()) > 0 {
+			// This directory node has children that have not yet been forgotten.
+			// We don't want to remove it from the tree (i.e. remove it from its parent)
+			// because the children may have parent pointers to it.
+			// A later LOOKUP will return this node with a new NodeID.
+			return
+		}
 		node.mount.treeLock.Lock()
-		c.recursiveConsiderDropInode(node)
+		// Remove Inode from all parents
```

Comment:
If it's not aware of tree structure, then what does it understand? The
lookup operation maps quite naturally to a nice containing a string indexed
map for its children.

Op di 22 mrt. 2016 22:32 schreef rfjakob <notifications@github.com>:

> In fuse/nodefs/fsconnector.go
> <https://github.com/hanwen/go-fuse/pull/103#discussion_r57073060>:
>
> >  		node.mount.treeLock.Lock()
> > -		c.recursiveConsiderDropInode(node)
> > +		// Remove Inode from all parents
>
> Thinking about it, probably nodefs should not beware of the tree structure
> at all? What do you think?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/hanwen/go-fuse/pull/103/files/004bb8a47f407fea15ba2caedcb216371ca806f4#r57073060>
>